src/services/backendService/index.js

```diff
@@ -0,0 +1,241 @@
+  muutospyynnot: `${API_BASE_URL}/muutospyynnot/`,
+  muutospyynto: `${API_BASE_URL}/muutospyynnot/id/`,
+  oivamuutoikeudetvelvollisuudetehdotjatehtavat: `${API_BASE_URL}/koodistot/koodit/oivamuutoikeudetvelvollisuudetehdotjatehtavat`,
+  opetuskielet: `${API_BASE_URL}/koodistot/opetuskielet`,
+  organisaatio: `${API_BASE_URL}/organisaatiot/`,
+  vankilat: `${API_BASE_URL}/koodistot/koodit/vankilat`,
+  liitteet: `${API_BASE_URL}/liitteet/`
+};
+
+/**
+ * Runs XHR calls.
+ *
+ * @param {string} key - A key of backendRoutes object.
+ * @param {string} url - XHR call is directed to this url.
+ * @param {function} dispatchFn - A dispatch function.
+ * @param {object} abortController - An instance of AbortController.
+ */
+async function run(
+  key,
+  url,
+  options,
+  dispatchFn,
+  abortController,
+  subKey,
+  path
+) {
+  /**
+   * It's time to fetch some data!
+   */
+  try {
+    const response = await fetch(url, {
+      ...options,
+      signal: abortController.signal
+    });
+
+    if (response && response.ok) {
+      const data = await response.json();
+
+      /**
+       * Data will be saved to the same context as what dispatchFn presents.
+       */
+      dispatchFn({
+        data,
+        key,
+        subKey,
+        path,
+        type: FETCH_FROM_BACKEND_SUCCESS
+      });
+    } else {
+      /**
+       * Fetching failed. So, let's mark it up for later use.
+       */
+      dispatchFn({
+        key,
+        response,
+        subKey,
+        path,
+        type: FETCH_FROM_BACKEND_FAILED
+      });
+    }
+    return response;
+  } catch (err) {
+    /**
+     * Fetching failed. So, let's mark it up for later use.
+     */
+    dispatchFn({
+      key,
+      err,
+      subKey,
+      path,
+      type: FETCH_FROM_BACKEND_FAILED
+    });
+  }
+}
+
+/**
+ *
+ * @param {Object[]} keysAndDispatchFuncs - [{ key: string, dispatchFn: function}, ...]
+ * @param {string} keysAndDispatchFuncs[].key - A key of the backendRoutes object.
+ * @param {string} keysAndDispatchFuncs[].dispatchFn - A dispatch function.
+ * @param {array} _abortControllers - Array of AbortController instances
+ * @param {number} index
+ */
+function recursiveFetchHandler(
+  keysAndDispatchFuncs,
+  _abortControllers = [],
+  _responses = [],
+  index = 0
+) {
+  const {
+    key,
+    dispatchFn,
+    options = {},
+    urlEnding = "",
+    subKey,
+    path
+  } = R.view(R.lensIndex(index), keysAndDispatchFuncs);
+
+  const url = R.join("", [R.prop(key, backendRoutes), urlEnding]);
+
+  /**
+   * When a user is leaving a page and the following backend call is active we can
+   * use an abort controller to gracefully abort the call. So, let's create one.
+   */
+  const abortController = new AbortController();
+
+  /**
+   * This is practically a push operation written with Ramda.
+   */
+  const abortControllers = R.insert(-1, abortController, _abortControllers);
```

Comment:
`R.append`?

Reply:
That's true. R.append is a little bit more appropriate solution. Will be implemented with R.append.